addons/web/static/src/public/public_component_service.js

```diff
@@ -0,0 +1,61 @@
+/** @odoo-module */
+
+import { App } from "@odoo/owl";
+import { registry } from "@web/core/registry";
+import { templates } from "@web/core/assets";
+import { _t } from "@web/core/l10n/translation";
+import { omit } from "@web/core/utils/objects";
+
+class ComponentManager {
+    constructor(env) {
+        this.env = env;
+        this.appConfig = {
+            templates,
+            env: env,
+            dev: env.debug,
+            translateFn: _t,
+            translatableAttributes: ["data-tooltip"],
+        };
+        /** @type {Map<HTMLElement, { app: App, mountProm: Promise<any> }>} */
+        this.apps = new Map();
+    }
+    mountComponents() {
+        for (const [key, component] of registry.category("public_components").getEntries()) {
+            for (const el of document.querySelectorAll(`[data-component="${CSS.escape(key)}"]`)) {
+                if (!this.apps.has(el)) {
```

Comment:
maybe not necessary as `data-component` can only be defined once and so cannot be targeted by multiple selector

Reply:
The purpose is to allow calling mountComponents again by hand if an element was added to the DOM with a [data-component] dynamically without respawning already instanciated components